### PR TITLE
Dragmap

### LINKED ISSUE #1106: new module: DRAGMAP
I think it would be good to have a module for [DRAGMAP](https://github.com/Illumina/dragmap)

- [X] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [X] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [X] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by Emiller88**
Closes #1106

### CODE REVIEW

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Very nice job @Emiller88 ! 
Just suggested whether it would be more optimal to output `bam` files directly (although this will need a mulled container 🙄) and also whether you plan to add the alt-masked bed file test on this iteration.

File: modules/dragmap/align/main.nf [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if (meta.single_end) {
+        """
+        dragen-os \\
+            -r $hashmap \\
+            -1 $reads \\
+            --output-directory . \\
+            --output-file-prefix $prefix \\
+            --num-threads $task.cpus \\
+            $args
+
```
[the remark this answers is not in the document]
Comment (Emiller88) [reply modules/dragmap/align/main.nf:30]: Yeah, I think that's what needs to happen. Same thing as the other aligners, to save on jobs, storage, etc.

File: modules/dragmap/align/meta.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        type: file
+        description: |
+            List of input FastQ files of size 1 and 2 for single-end and paired-end data,
+            respectively.
+    - hashmap:
+        type: file
+        description: DRAGMAP hash table
+        pattern: "Directory containing DRAGMAP hash table *.{cmp,.bin,.txt}"
+output:
+    - sam:
```
Comment (JoseEspinosa) [inline modules/dragmap/align/meta.yml:33]:
```suggestion
    - bam:
```

**Review by JoseEspinosa — APPROVED** [review]
Awseome @Emiller88 ! 🚀

**Pull request merged**